consensus/obcpbft/obc-classic.go

```diff
@@ -129,23 +129,25 @@ func (op *obcClassic) validate(txRaw []byte) error {
 
 // execute an opaque request which corresponds to an OBC Transaction
 func (op *obcClassic) execute(seqNo uint64, txRaw []byte) {
-	tx := &pb.Transaction{}
-	err := proto.Unmarshal(txRaw, tx)
-	if err != nil {
-		logger.Error("Unable to unmarshal transaction: %v", err)
-		return
-	}
-
-	meta, _ := proto.Marshal(&Metadata{seqNo})
-
-	id := []byte("foo")
-	op.stack.BeginTxBatch(id)
-	result, err := op.stack.ExecTxs(id, []*pb.Transaction{tx})
-	_ = err    // XXX what to do on error?
-	_ = result // XXX what to do with the result?
-	_, err = op.stack.CommitTxBatch(id, meta)
-
-	op.pbft.execDone()
+	go func() {
+		tx := &pb.Transaction{}
+		err := proto.Unmarshal(txRaw, tx)
+		if err != nil {
+			logger.Error("Unable to unmarshal transaction: %v", err)
+			return
+		}
+
+		meta, _ := proto.Marshal(&Metadata{seqNo})
+
+		id := []byte("foo")
+		op.stack.BeginTxBatch(id)
+		result, err := op.stack.ExecTxs(id, []*pb.Transaction{tx})
+		_ = err    // XXX what to do on error?
```

Comment:
Why would we roll back the entire batch of transactions if we encountered a single error?

There has been some substantial debate about this, but today, the path is that we record all the transactions, and accompanying errors in the block.  There has been the argument, that we should not recorded errored transactions, but this should be implemented away from consensus.  (It is currently in helper, which is generally an odd place to be making decisions about the ledger contents).

In general, I agree that we need a stronger contract from the consensus API, "when an error is received, it means X and you should do Y", but today, it's not clear that rolling back is correct, and if we were to rollback, what should be the next step? Should we retry?  Should we discard? Should we panic? There's no clear recovery path, which is why this is flagged in the code as a trouble spot.

Please note, that this is not new logic for this PR.